### PR TITLE
Modify code to support converting to C++

### PR DESCRIPTION
**Pull request by marsfan**
This is a first pass. There is still work to be done, but it can be converted to C++ at this point. I had to initialize a lot of variables. 
The file `RobotControl.prj` is the code generation project. It is already configured and ready to go. 
The file `codeGenConfig.mat` contains a backup of the configuration settings for MATLAB Coder. Use it to restore the settings. 

<details>
<summary> Remaining work </summary>

* Constant code folding has to remain at default. If lowered, it breaks some internal MATLAB Functions :(
* Initialize function has to be enabled.

Optimization to still take
* Disable constant folding (as we can trust G++ to handle this)
* Disable both initialize and terminate functions
* Disable support for variable array size (Requires fixing mu and removing all calls to find())
* Figure out how to enable support for NEON
* Enable inlining? ((Or not, should we trust GCC?)
* Disable support for nonfinite numbers (Seems to be find)
* Manually type variables to reduce use of doubles.
* Optimize Robot_Control to be more linear (less undefined var from branch)
* Optimize Robot_Control to avoid persistent variables
* Remove unneeded parts from Robot_Control
* Enable MathWorks typedefs for MISRA Compliance
* Investigate potential differences
* Fix all TODOs
* Identify large CPP/H files and re-write to shrink. This includes (but is not limited to)
  * coder_array.h
  * inpolygon.cpp
  * rtwtypes.h
  * sortldx.cpp
  * xzsvdc.cpp
  * Make sure codegen is not skipping unused paths, as they later might be used.
  
  </details>

### CODE REVIEW

File: SpringProject/kinematic_controller_solver_test.m [hunk 1]
```diff
@@ -124,6 +124,7 @@
 phi_d = pi/2.*ones(1,length(x_d));
 theta_d =  zeros(1,length(x_d));
 psi_d =  zeros(1,length(x_d));
+state = zeros(1, 36);
```
Comment (nicholasvrossi) [inline SpringProject/kinematic_controller_solver_test.m:127]: switch indexing dimensions

File: SpringProject/manipulability_test.m [hunk 2]
```diff
@@ -42,6 +42,7 @@
 % draw robot to confirm (FK)
 
 
+state = zeros(1,36);
```
Comment (nicholasvrossi) [inline SpringProject/manipulability_test.m:45]: switch indexing dims

File: SpringProject/robot_controller_v3.m [hunk 3]
```diff
@@ -128,6 +128,7 @@
 Kp_floor = -5000; % floor spring constant
 Kd_floor = -800; % floor damping coefficient
 b_fric_floor = -2000; % floor coefficient of lateral, viscous friction
+state = zeros(1,36);
```
Comment (nicholasvrossi) [inline SpringProject/robot_controller_v3.m:131]: and here

File: SpringProject/robot_solver_split_controller_V2.m [hunk 4]
```diff
@@ -97,6 +97,7 @@
 Kp_floor = -5000; % floor spring constant
 Kd_floor = -800; % floor damping coefficient
 b_fric_floor = -2000; % floor coefficient of lateral, viscous friction
+state = zeros(1, 36);
```
Comment (nicholasvrossi) [inline SpringProject/robot_solver_split_controller_V2.m:100]: bazinga

**Review by nicholasvrossi — APPROVED** [review]

File: SpringProject/kinematic_controller_solver_test.m (hunk 1 above)
Comment (marsfan) [reply SpringProject/kinematic_controller_solver_test.m:127]: Fixed in c0609a5fd791bf6ff06dc30c0050992cd16cbaa7

File: SpringProject/robot_controller_v3.m (hunk 3 above)
Comment (Marsfan) [reply SpringProject/robot_controller_v3.m:131]: fixed in ce27c1abd3bb5b6b2af4460e46af67cfc407ce39

File: SpringProject/manipulability_test.m (hunk 2 above)
Comment (marsfan) [reply SpringProject/manipulability_test.m:45]: fixed in ce27c1abd3bb5b6b2af4460e46af67cfc407ce39

File: SpringProject/robot_controller_v3.m (hunk 3 above)
Comment (marsfan) [reply SpringProject/robot_controller_v3.m:131]: Fixed in 7d35f0b138ebeaaf6bb62ae2fa4fc652f9b7e359

File: SpringProject/robot_solver_split_controller_V2.m (hunk 4 above)
Comment (marsfan) [reply SpringProject/robot_solver_split_controller_V2.m:100]: fixed in 5a5ef73d32e69b13da3e458b634e3e16500b227e

**Pull request merged**